nixos/modules/services/databases/lldap.nix

```diff
@@ -149,6 +187,8 @@ in
             default = null;
             description = ''
               Path to a file containing the JWT secret.
+
+              **Deprecated**: Use `credentials.LLDAP_JWT_SECRET_FILE` instead.
```

Comment:
We can't do these deprecations. These are defined upstream in their config file format, and we are just documenting the options, that we're interacting with in the module.

In fact, most (all?) of lldap's supported environment variables are auto-mapped from these options, and I'd consider the config file option to be the primary source. (Your decision to prefer the env var is still correct though, IMO)

See other uses of `freeformType` to get an idea how this works.

What you can do, is add warnings (or assertions in severe cases) where setting these option would interfere with our credential handling. In fact some of them are only documented, so we can mention them in warnings.